pkg/registry/apps/daemonset/strategy.go

```diff
@@ -66,6 +67,25 @@ func (daemonSetStrategy) NamespaceScoped() bool {
 	return true
 }
 
+// GetResetFields returns the set of fields that get reset by the strategy
+// and should not be modified by the user.
+func (daemonSetStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
+		"apps/v1": fieldpath.NewSet(
+			fieldpath.MakePathOrDie("status"),
+			fieldpath.MakePathOrDie("spec", "templateGeneration"),
+		),
+	}
+
+	// TODO: make sure this works and is the best way to do this
```

Comment:
TODO: make sure template fields are added correctly
@apelisse do you know a better way to do this? Is this even correct?